File: babylonian_square_root/babylonian_square_root.js
function squareRoot(num) {
  const e = 0.00000001;
  let x = num;
  let y = 1;

  while(x - y > e) {
    x = (x + y) / 2;
    y = num / x;
  }

  return x;
}

console.log(squareRoot(5));

File: babylonian_square_root/babylonian_square_root.cpp
#include <iostream>
#include <iomanip>
using namespace std;

double square_root(double num) {

	double x = num;
	double y = 1;
	double e = 0.00000001;

	while (x - y > e) {
		x = (x + y) / 2;
		y = num / x;
	}
	return x;
}

int main() {

	cout << setprecision(10) << square_root(6);
	return 0;
}
